R/textstat_simil.R

```diff
@@ -358,3 +379,37 @@ as.matrix.simil <- function(x, diag = 1.0, ...) {
     diag(df) <- diag
     df
 }
+
+#' Coerce a textstat_simil or textstat_dist to a sparse Matrix format
+#' 
+#' Coerce a \link{textstat_simil} or \link{textstat_dist} to a sparse \pkg{Matrix} format.
+#' @param x input textstat object
+#' @keywords textstat internal
+#' @export
+as.Matrix <- function(x) {
+    UseMethod("as.Matrix")
+}
+
+#' @export
+#' @method as.Matrix simil_pairwise
+#' @keywords textstat internal
+as.Matrix.simil_pairwise <- function(x) {
+    sparseMatrix(i = as.integer(x$x), j = as.integer(x$y),
+                 x = x$similarity,
+                 dims = c(nlevels(x$x), nlevels(x$y)),
+                 dimnames = list(levels(x$x), levels(x$y)))
+}
+
+#' @export
+#' @method as.matrix simil_pairwise
+#' @keywords textstat internal
+as.matrix.simil_pairwise <- function(x, ...) {
+    as.matrix(as.Matrix(x))
+}
+
+#' @export
+#' @method as.list simil_pairwise
+#' @keywords textstat internal
+as.list.simil_pairwise <- function(x, ...) {
```

Comment:
Let's keep the full existing `as.list()` functionality, regardless of the value of `tri/upper`, i.e.
```r
as.list.simil_pairwise(x, sorted = TRUE, n = NULL, ...)
```

Reply:
Users can sort the output just by `tstat2[order(tstat2$similarity),]` because it is a `data.frame`

Reply:
Fine for the data.frame, but why change the behaviour of `as.list()`?

Reply:
Restored the previous behaviour, although if `diag = FALSE`, then the coercion to list omits identity pairs.

5bf1269